lib/client.js

```diff
@@ -131,7 +211,16 @@ function flushVolatile (queue) {
 
 function storeAndSend (client, packet, cb, cbStorePut) {
   debug('storeAndSend :: store packet with cmd %s to outgoingStore', packet.cmd)
-  client.outgoingStore.put(packet, function storedPacket (err) {
+  var storePacket = packet
+  var err
+  if (storePacket.cmd === 'publish') {
+    storePacket = clone(packet)
```

Comment:
add comment explaining that you are cloning the packet because removeTopicAlias mutates the object passed in to it.

Reply:
This function is `storeAndSend`.
At the storing process, Topec Alias needs to be removed and Topic Name needs to recovered for the re-sending.
But the original packet should be sent. So I need to clone.